### PR TITLE
Add scitokens

### PR DESCRIPTION
**Pull request by djw8605**
This pull request adds scitokens configuration for authenticated and non-authenticated caches and origins.

### CODE REVIEW

**Comment by djw8605** [comment]
@matyasselmeci this has been "Development" tested, I'm ready to move to the next step of building scratch RPMs and testing full installs.

**Comment by brianhlin** [comment]
As I mentioned in the #xcache Slack, since the SciTokens integration requires XRootD 5 can we create a new branch for this so we can carry an XCache 2.0 in upcoming while maintaining 1.x for the mainline release?

**Comment by djw8605** [comment]
It doesn't "require" xrootd 5.  It is certainly functional now.  But, the data and credentials are transferred in the clear (albeit, in xrootd binary protocol) between the cache and origins.

**Comment by brianhlin** [comment]
> But, the data and credentials are transferred in the clear (albeit, in xrootd binary protocol) between the cache and origins.

Yeah, then this seems like something we should be doing in a separate branch that's tied to XRootD 5/upcoming.

**Comment by bbockelm** [comment]
I’m OK with branching this out to force encryption - especially as this should be possible with the RC.

**Comment by djw8605** [comment]
@brianhlin I am ready to merge this since XRootD 5 is out.

File: configs/stash-cache/config.d/50-stash-cache-authz.cfg [hunk 1]
```diff
@@ -64,3 +64,8 @@ fi
 # the OSG packaging; see SOFTWARE-3490.
 xrootd.diglib * /etc/xrootd/digauth.cfg
 
+# Allow scitokens on all ports, all protocols
+ofs.authlib libXrdAccSciTokens.so config=/run/stash-cache-auth/scitokens.conf
+# Pass the bearer token to the Xrootd authorization framework.
+http.header2cgi Authorization authz
+
```
Comment (brianhlin) [inline configs/stash-cache/config.d/50-stash-cache-authz.cfg:71]:
```suggestion
```

File: configs/stash-origin/config.d/50-stash-origin-authz.cfg [hunk 2]
```diff
@@ -60,3 +60,8 @@ else if named stash-origin
    acc.authdb /run/stash-origin/Authfile
 fi
 
+# Allow scitokens always, whether auth origin or not.
+ofs.authlib libXrdAccSciTokens.so config=/run/stash-origin-auth/scitokens.conf
+
+
+
```
Comment (brianhlin) [inline configs/stash-origin/config.d/50-stash-origin-authz.cfg:67]:
```suggestion
```

**Comment by djw8605** [comment]
Performed the requested changes.

**Pull request merged**